projects/output-and-error-handling/parseRetryAfterString_test.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestParseRetryAfterString(t *testing.T) {
+	t.Run("Changing a timestamp as a string to time.Duration - using parse ", func(t *testing.T) {
+		timestampStr := time.Now().UTC().Add(time.Duration(4) * time.Second).Format(http.TimeFormat)
+		timestampParse, _ := time.Parse(http.TimeFormat, timestampStr)
+		got := parseRetryAfterString(timestampStr)
+		want := time.Until(timestampParse)
+
+		if got != want {
+			t.Errorf("got %s want %d", got, want)
+		}
+	})
+
+	t.Run("Changing a timestamp as a string to time.Duration - without pars", func(t *testing.T) {
+		timestampStr := time.Now().UTC().Add(time.Duration(4) * time.Second).Format(http.TimeFormat)
+		got := parseRetryAfterNumber(timestampStr)
+
+		if got < 3*time.Second || got > 4*time.Second {
```

Comment:
This is interesting! Why did you have to use a range here rather than an exact value?

Reply:
I wanted to test the result without using those methods that I used in the function to make sure I chose correct ones? At the moment I did with checking the range but I need to find the exact value in another way.